flight/PiOS.posix/posix/pios_led.c

```diff
@@ -28,19 +28,53 @@
 
 /* Project Includes */
 #include "pios.h"
+#include "openpilot.h"
 
 #if defined(PIOS_INCLUDE_LED)
 
 /* Private Function Prototypes */
 
+DONT_BUILD_IF(PIOS_LED_NUM > 8, piosLedNumTooHigh);
 
 /* Local Variables */
-static uint8_t LED_GPIO[PIOS_LED_NUM];
+static uint8_t ledStatus;
 
+static inline void PIOS_SetLED(uint32_t led, uint8_t stat) {
+	uint8_t newStatus = ledStatus;
 
-static inline void PIOS_SetLED(uint32_t LED,uint8_t stat) {
-	printf("PIOS: LED %i status %i\n",LED,stat);
-	LED_GPIO[LED]=stat;
+	PIOS_Assert(led < PIOS_LED_NUM);
+
+	uint8_t mask = 1<<led;
+
+	if (stat) {
+		newStatus |= mask;
+	} else {
+		newStatus &= ~mask;
+	}
+
+	if (newStatus != ledStatus) {
+		ledStatus = newStatus;
+
+		char leds[PIOS_LED_NUM+1];
+
+		uint8_t mask = 1;
+
+		for (int i=0; i<PIOS_LED_NUM; i++) {
+			if (newStatus & mask) {
+				leds[i] = '*';
+			} else {
+				leds[i] = '.';
+			}
+
+			mask <<= 1;
+		}
+
+		leds[PIOS_LED_NUM] = 0;
+
+		printf("PIOS: LEDs: [%s]\n", leds);
+	}
+
+	//printf("PIOS: LED %i status %i\n",led,stat);
```

Comment:
I think leaving the obvious debug line in a subsystem saves anyone who has to investigate it in the future some time.

If no one comes back here, it can be safely forgotten.  If it annoys the next guy, it's trivial to remove and isn't cluttering things measurably  (hopefully he removes it if it's no longer applicable).  If it's helpful, well, it's helpful.